HIRING.md

```diff
@@ -0,0 +1,52 @@
+# Interviewing for an iOS position
+
+## Overview
+This document provides an outline of how you can structure an interview. It also contains some questions to ask as well as some possible technical challenges to ask.
+
+## Purposes of an interview
+We’re obviously trying to assess the candidate but remember they are assessing us. As much as possible an interview should be a two-way conversation; an opportunity to talk with a peer about some technical subjects. It is also a marketing opportunity -- make sure the candidate goes away with a positive impression of B&B -- even if you know the candidate is not going to be hired after 5 minutes ensure they have a good time for an hour. They may reapply later on; or they may recommend us to their better suited friend.
+
+## Structure
+
+### Use about 5 minutes
+- Introduce yourself
+- Introduce B&B (Office locations, what we do)
+- Introduce the position they are interviewing for
+
+### Use about 10-15 minutes
+- Pick a role from their CV that looks interesting (or their final project if a graduating student)
+- Ask them to talk about the role; their responsibilities; things that went well; things that went badly; why they went badly. Here you are trying to assess how they are as a worker -- are they a leader, or a follower? Are they outspoken or quiet? Do they work well in a team, or alone? You are also finding out how good they are at communicating -- can you understand what they actually did; can they answer clarifying questions about what they did? This has the benefit of warming up the candidate; talking about something they have done is easy.
+
+### Use about 20-30 minutes
+- Ask a few technical questions. Examples are found below.
+- Here you are trying to assess what they know. At the same time you can assess if they are honest in what they don’t know. Push them to the limits of their knowledge but don’t waste time if they don’t know something; change your questioning to their knowledge so that they are suitably challenged.
```

Comment:
This can be so very tricky. Hard not to come off as hostile or exacerbate impostor syndrome.

Reply:
I am late to this party! This whole hiring page was based off of a document I wrote for Android - so these words are my own.

Let me explain.

There is zero point asking a candidate about things they don't know. It's a waste of everyones time and doesn't make for a pleasant conversation.

Asking about what they know well is useful, but it doesn't tell you much more than that.

Instead, push them to the limits of what they know -- find where that limit is and stay on it. You'll find out more about the candidate. Are they curious? Do they have good instincts? Can they guess about things they don't know? Do they ask us interesting or clarifying questions? Do they go quiet and get frustrated? If you have to help them, how do they react? Are they good at communicating things they're uncertain about?

If you sense that the candidate is struggling, help them out a bit and bring it a little closer to their comfort zone.

It can be difficult to find this line, but by starting with something they know and drilling down into how much they know about that thing - you'll find a line eventually. All the time though, it should be a conversation - not an interrogation. We do this all the time as devs, we have discussions about things we are uncertain about; and we discuss and make plans around our uncertainty.